scripts/can_errgo/error_tui.go

```diff
@@ -0,0 +1,668 @@
+	canFrameSize = 16       // CAN frame size (for a standard CAN frame)
+)
+
+// Might have future use
+func deepCopy(rows []table.Row) []table.Row {
+    newRows := make([]table.Row, len(rows))
+    for i := range rows {
+        newRows[i] = make(table.Row, len(rows[i]))
+        copy(newRows[i], rows[i])
+    }
+    return newRows
+}
+
+// Converts a 8-byte slice to a uint64, this is for masking purposes.
+func bytesToUint64(b []byte) uint64 {
+    var result uint64
+    for i := 0; i < len(b) && i < 8; i++ {
+        // Reverse the order: least significant byte (rightmost) gets the lowest bits
+        result |= uint64(b[len(b)-1-i]) << (8 * i)
+    }
+    return result
+}
+
+var baseStyle = lipgloss.NewStyle().
+	BorderStyle(lipgloss.NormalBorder()).
+	BorderForeground(lipgloss.Color("240"))
+
+
+type CANMsg struct {
+	ID    uint32
+	Value uint64
+}
+
+// Empty type for the tick message
+type TickMsg struct{}
+
+type model struct {
+	// Main table
+    table        table.Model
+	tableKeys	 KeyMap	
+
+	// Submenu table
+	submenuTable  table.Model
+	submenuActive bool
+	submenuKeys   KeyMap
+
+	// Hidden error counts and error to bit mapping
+    hiddenCounts map[string]int
+    errorToBit   map[string]int
+    ignoreMask   uint64
+
+	// Timeout flag and last CAN message time
+	isTimeout   bool
+	lastCANTime time.Time
+	t           time.Duration
+
+	// Box for displaying error descriptions
+	box box.Box
+	showBox bool
+}
+
+
+// toggleBit toggles the specified bit in the number
+//If the bit is 0, it sets it to 1; if it is 1, it sets it to 0.
+func toggleBit(n uint64, bitPosition int) uint64 {
+    mask := uint64(1) << bitPosition
+    return n ^ mask
+}
+
+
+func deleteElementRow(slice []table.Row, index int) []table.Row {
+    // Create a new slice with the same length minus 1
+    newSlice := make([]table.Row, 0, len(slice)-1)
+                      
+    // Append the elements before the index
+    newSlice = append(newSlice, slice[:index]...)
+
+    // Append the elements after the index
+    newSlice = append(newSlice, slice[index+1:]...)
+
+    return newSlice
+}
+
+func tickEvery(t time.Duration) tea.Cmd {
+	return func() tea.Msg {
+		time.Sleep(t)
+		return TickMsg{}
+	}
+}
+
+// Might have future use
+func findRow(s []table.Row, e table.Row) int {
+    for i, a := range s {
+        if a[0] == e[0] {
+            return i
+        }
+    }
+    return -1
+}
+
+// Find the index of a row with a specific error name.
+func findRowString(s []table.Row, e string) int {
+    for i, a := range s {
+        if a[0] == e {
+            return i
+        }
+    }
+    return -1
+}
+
+// Check if the cursor is over bounds
+func overBounds(table table.Model) bool{
+	return len(table.Rows())==0 || table.Cursor() < 0 || table.Cursor() >= len(table.Rows()) 
+}
+
+
+func (m *model) re_fresh(){
+	rows := m.table.Rows()
+	for i:=0; i < len(rows); i++{
+		if val,err := strconv.Atoi(rows[i][2]); err != nil{
```

Comment:
the "freshness" should be stored with the error message struct, not in the table. It is a "code smell" to see this value being stored in the table text and converted to & from string